document.go

```diff
@@ -388,6 +398,9 @@ func newDocument(v interface{}, rv reflect.Value, readonly bool) *Document {
 		}
 		rt = rt.Elem()
 	}
+	if rt.Kind() == reflect.Ptr {
```

Comment:
This was removed in your PR, without this that test case panics

Reply:
what if the pointer contains pointer to nil?

Reply:
wrong test case, sorry 🤦🏻‍♂️